src/kx122/kx122.hpp

```diff
@@ -0,0 +1,505 @@
+* Copyright (c) 2018 Rohm Semiconductor.
+*
+* Permission is hereby granted, free of charge, to any person obtaining a copy of
+* this software and associated documentation files (the "Software"), to deal in
+* the Software without restriction, including without limitation the rights to
+* use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+* the Software, and to permit persons to whom the Software is furnished to do so,
+* subject to the following conditions:
+*
+* The above copyright notice and this permission notice shall be included in all
+* copies or substantial portions of the Software.
+*
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+* FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+* COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+* IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+* CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+#include <mraa/gpio.hpp>
+
+#include "kx122.h"
+
+/**
+ * @file kx122.hpp
+ * @library kx122
+ * @brief C++ API for the kx122 driver
+ *
+ * @include kx122.cxx
+ */
+
+namespace upm{
+
+  class KX122{
+    public:
+
+      /**
+      KX122 constructor
+      Set addr to -1 if using SPI.
+      When using I2C, set chip_select_pin to -1
+
+      If no errors occur, the device gets initialized with default values and gets set to active state.
+
+      @param bus I2C or SPI bus to use.
+      @param addr I2C address of the sensor.
+      @param chip_select Chip select pin for SPI.
+      @throws std::runtime_error on initialization failure.
+      */
+      KX122(int bus, int addr, int chip_select);
+
+      /**
+      KX122 destructor
+      */
+      ~KX122();
+
+      /**
+      Initializes the sensor with given sampling rate, resolution and acceleration range.
+      This gets called in the constructor, so it will not need to be called unless the device is reset.
+
+      Sensor is automatically set into standby mode during the initialization.
+      Sensor is set to active mode after initialization.
+
+      @param odr One of the KX122_ODR_T values.
+      @param res One of the KX122_RES_T values.
+      @param grange One of the KX122_RANGE_T values.
+      @throws std::runtime_error on failure.
+      */
+      void deviceInit(KX122_ODR_T odr, KX122_RES_T res, KX122_RANGE_T grange);
+
+      /**
+      Gets the length of one sample period depeding on the sampling rate of the sensor.
+      @return Floating point value of the sampling period, or if error occurs returns -1.
+      */
+      float getSamplePeriod();
+
+      /**
+      Gets who am i value of the sensor.
+
+      @param data Pointer to a uint8_t variable to store the value.
+      @throws std::runtime_error on failure.
+      */
+      void getWhoAmI(uint8_t *data);
```

Comment:
Double-check this - I'm not seeing the implementation for getWhoAmI anywhere.

Reply:
Interesting that i somehow managed to completely miss that.
Anyways, the implementation is now added.